lib/logstash/outputs/elasticsearch.rb

```diff
@@ -368,9 +370,31 @@ def submit(actions)
   def flush(actions, teardown = false)
     begin
       submit(actions)
+    rescue Manticore::SocketException => e
+      # If we can't even connect to the server let's just print out the URL (:host is actually a URL)
+      # and let the user sort it out from there
+      @logger.error(
+        "Attempted to send a bulk request to Elasticsearch configured at '#{@current_client.client_options[:host]}',"+
+          " but Elasticsearch appears to be unreachable or down!",
+        :client_config => @current_client.client_options,
+        :error_message => e.message,
+        :error_class => e.class.name,
+        :backtrace => e.backtrace
+      )
+      @logger.debug("Failed actions for last bad bulk request!", actions: actions)
     rescue => e
-      @logger.error "Got error to send bulk of actions: #{e.message}"
-      raise e
+      # For all other errors print out full connection issues
+      @logger.error(
+        "Attempted to send a bulk request to Elasticsearch configured at '#{@current_client.client_options[:host]}'," +
+            " but an error occurred and it failed! Are you sure you can reach elasticsearch from this machine using " +
```

Comment:
indentation changes across these 3 lines of string. Expected?

Also, for long messages like this, we should consider putting them in `locales/en.yml` and looking them up with I18n.t(...) - thoughts?